### ISSUE TITLE
[Explainability Evaluation] - GNN model (Un)Faithfulness

### ISSUE THREAD

**Issue by BlazStojanovic** [pyg-team/pytorch_geometric#5961]
### 🚀 The feature, motivation and pitch

Implement **faithfulness** [1],  an explainability metric which quantifies how faithful the explanations are to an underlying GNN predictor. This feature request is a part of broader explainability evaluation metrics (parent issue https://github.com/pyg-team/pytorch_geometric/issues/5628) which will be crucial for the new revamp of explainability (see roadmap https://github.com/pyg-team/pytorch_geometric/issues/5520).

### Overview of the metric
**Un-Faithfulness** of a GNN predictor is given as 
$$\operatorname{GEF}\left(f, \mathcal{S}_u, \mathcal{S}_u^{\prime}\right)=1-\exp ^{-\mathrm{KL}\left(f\left(\mathcal{S}_u\right) \| f\left(\mathcal{S}_u^{\prime}\right)\right)}, $$
where:
* $\hat y_u$ is the prediction probability vector obtained by using the GNN $\hat{y}_u=f\left(\mathcal{S}_u\right)$
* $\hat y_{u^\prime}$ is also obtained with the GNN as $\hat{y}_{u^{\prime}}=f \left(\mathcal S_u ^ \prime \right)$, where $\mathcal S_u^\prime$ is a masked subgraph generated by only keeping the original values of the top-k features identified by an explanation, and
* $KL(\cdot || \cdot)$ is the [Kullback–Leibler divergence](https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence)

### Implementation checklist
* Implement a method that computes the unfaithfulness of a GNN predictor

### References
[1] - [Evaluating Explainability for Graph Neural Networks](https://arxiv.org/abs/2208.09339)
